### PR TITLE
Octave: use correct mkoctfile executable

### PR DESCRIPTION
**Pull request by mtmiller**
Allow a versioned Octave executable to be configured via --with-octave
and the correct corresponding mkoctfile executable to be used.

### CODE REVIEW

**Comment by wsfulton** [comment]
Another one for @kwwette to look at.

**Comment by kwwette** [comment]
This looks fine to me.

**Comment by wsfulton** [comment]
@mtmiller, would you mind giving some details of the files that this now fixes. I'm finding it hard to follow without the particular version of Octave installed.

**Comment by mtmiller** [comment]
@wsfulton, this fix allows swig to be configured to a specific Octave if more than one version are co-installed. For example I have `/opt/gnu/octave/bin/octave-3.8.2` and `/opt/gnu/octave/bin/octave-4.0.0`.

Without this change, as it is currently, `./configure --with-octave=$mybindir/octave-3.8.2` will set `mkoctfile=$mybindir/mkoctfile`, which may or may not be the right version for the Octave I want to use.

With this change, we have:


`./configure --with-octave=` | `mkoctfile=`
--- | ---
`$dir/octave` | `$dir/mkoctfile`
`$dir/octave-3.8.2` | `$dir/mkoctfile-3.8.2`
`$dir/octave-4.0.0` | `$dir/mkoctfile-4.0.0`

I think this fix continues to support the most common case of `/usr/bin/octave` and `/usr/bin/mkoctfile` as well as allowing developers like me to pick from multiple versions of Octave in parallel.
